src/Equinor.ProCoSys.IPO.Command/InvitationCommands/CreateInvitation/CreateInvitationCommandHandler.cs

```diff
@@ -66,20 +70,45 @@ public async Task<Result<int>> Handle(CreateInvitationCommand request, Cancellat
             }
 
             participants = await AddParticipantsAsync(invitation, participants, request.Participants.ToList());
+            await _unitOfWork.SaveChangesAsync(cancellationToken);
 
             try
             {
-                var meetingId = await CreateOutlookMeeting(request, participants);
+                var meetingId = await CreateOutlookMeeting(request, participants, invitation);
                 invitation.MeetingId = meetingId;
+                await _unitOfWork.SaveChangesAsync(cancellationToken);
             }
             catch
             {
+                UndoSaveChanges(invitation);
```

Comment:
@paal123

Reply:
@bratli I suggest we try to rather use
using (var transaction = await context.Database.BeginTransactionAsync())

Around the whole method content  here. Then we can do savechanges to the get the db id, and the commit after we have called Fusion. Once fusino has reponded we doa Commit() and we should be fine.


If a problem occurs we can just do transaction.Rollback() and all changes in the db is cleaned up automatically.

You can also see more details here https://docs.microsoft.com/en-us/ef/core/saving/transactions